spinoffs/The_Beautiful_Child/scenarios/Valley_Passage.cfg

```diff
@@ -890,50 +885,63 @@
                         [store_unit]
                             [filter]
                                 side=1
-                                [filter_wml]
-                                    [movement_costs]
-                                        mountains={UNREACHABLE}
-                                    [/movement_costs]
-                                [/filter_wml]
                             [/filter]
-                            variable=cant_pass
+                            variable=side1
                         [/store_unit]
+                        {VARIABLE changed_mnt_cost no}
                         [for]
-                            array=cant_pass
+                            array=side1
                             [do]
-                                [set_variables]
-                                    name=cant_pass[$i].modifications.advancement[$cant_pass[$i].modifications.advancement.length]
-                                    mode=replace
-                                    [value]
-                                        id=mountain_passing
-                                        [effect]
-                                            apply_to=movement_costs
-                                            replace=true
-                                            [movement_costs]
-                                                mountains=3
-                                            [/movement_costs]
-                                        [/effect]
-                                    [/value]
-                                [/set_variables]
-                                {VARIABLE cant_pass[$i].movement_costs.mountains 3}
-                                [unstore_unit]
-                                    variable=cant_pass[$i]
-                                    find_vacant=no
-                                [/unstore_unit]
+                                [if]
+                                    [variable]
+                                        name=side1[$i].movement_costs.mountains
+                                        greater_than=0    # not null
+                                    [/variable]
+                                    [and]
+                                        [variable]
+                                            name=side1[$i].movement_costs.mountains
+                                            less_than=4
+                                        [/variable]
+                                    [/and]
```

Comment:
Units for whom it's null should pass too.

Reply:
If it's null, the unit is modified by the else block.  I need that check because null is considered less than 4.

I wonder if x=="" is equivalent to x==NULL.  But then I've seen you use this if/else format enough I assumed you were doing it for a reason (e.g. if x>3 else... is basically if x != NULL and X<=3 then ...)

I don't describe that well, but this works.  I tested it a lot.

Reply:
All right.